rocAL/rocAL/source/meta_data/bounding_box_graph.cpp

```diff
@@ -210,89 +203,89 @@ void BoundingBoxGraph::update_box_encoder_meta_data(std::vector<float> *anchors,
             {
                 if (offset)
                 {
-                    encoded_bb[anchor_idx] = {0, 0, 0, 0};
+                    encoded_bb[anchor_idx].xcycwh = {0, 0, 0, 0};
                     encoded_labels[anchor_idx] = 0;
                 }
                 else
                 {
                     //Convert the "ltrb" format to "xcycwh"
-                    encoded_bb[anchor_idx].xc = 0.5 * (p_anchor->l + p_anchor->r); //xc
-                    encoded_bb[anchor_idx].yc = 0.5 * (p_anchor->t + p_anchor->b); //yc
-                    encoded_bb[anchor_idx].w = (-p_anchor->l + p_anchor->r);      //w
-                    encoded_bb[anchor_idx].h = (-p_anchor->t + p_anchor->b);      //h
+                    encoded_bb[anchor_idx].xcycwh.xc = 0.5 * (p_anchor->ltrb.l + p_anchor->ltrb.r); //xc
+                    encoded_bb[anchor_idx].xcycwh.yc = 0.5 * (p_anchor->ltrb.t + p_anchor->ltrb.b); //yc
+                    encoded_bb[anchor_idx].xcycwh.w = (-p_anchor->ltrb.l + p_anchor->ltrb.r);      //w
+                    encoded_bb[anchor_idx].xcycwh.h = (-p_anchor->ltrb.t + p_anchor->ltrb.b);      //h
                     encoded_labels[anchor_idx] = 0;
                 }
             }
         }
-        BoundingBoxCords * encoded_bb_ltrb = (BoundingBoxCords*)&encoded_bb;
-        full_batch_meta_data->get_bb_cords_batch()[i] = (*encoded_bb_ltrb);
-        full_batch_meta_data->get_labels_batch()[i] = encoded_labels;
+
+        memcpy(encoded_boxes_data + (i * anchors_size * 4) , encoded_bb.data(), sizeof(float) * 4 * anchors_size);
```

Comment:
This memcopy can be avoided by making it to directly point to encoded_bb, same for labels memcopy